cmd/end2endtest/main.go

```diff
@@ -172,20 +172,61 @@ func NewAPIclient(host string) (*apiclient.HTTPclient, error) {
 	if err != nil {
 		log.Fatal(err)
 	}
-
 	log.Debugf("connecting to %s", hostURL.String())
-
 	token := uuid.New()
+
 	return apiclient.NewHTTPclient(hostURL, &token)
 }
 
-func privKeyToSigner(key string) (*ethereum.SignKeys, error) {
-	var skey *ethereum.SignKeys
-	if len(key) > 0 {
-		skey = ethereum.NewSignKeys()
-		if err := skey.AddHexKey(key); err != nil {
-			return nil, fmt.Errorf("cannot create key %s with err %s", key, err)
+func main() {
+	fmt.Fprintf(os.Stderr, "vocdoni version %q\n", internal.Version)
+
+	c := &config{}
+	parseFlags(c)
+	initializeLogger(c)
+
+	if len(c.accountPrivKeys) == 0 {
+		c.accountPrivKeys = []string{util.RandomHex(32)}
+		log.Infof("no keys passed, generated random private key: %s", c.accountPrivKeys)
+	}
+
+	if err := createSignKeys(c); err != nil {
+		log.Fatal(err)
+	}
+
+	api, err := NewAPIclient(c.host)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	if c.operation == allTests {
+		wg := &sync.WaitGroup{}
+		for _, op := range ops {
+			wg.Add(1)
+			go func(op operation) {
+				defer wg.Done()
+
+				cfgCopy := *c
+				cfgCopy.accountPrivKeys = []string{util.RandomHex(32)}
+				if err := createSignKeys(&cfgCopy); err != nil {
+					log.Fatal(err)
+				}
+				c = &cfgCopy
+				if err := setupAndRun(op, api, c); err != nil {
+					log.Fatal(err)
+				}
+			}(op) // Pass the op variable
```

Comment:
you're mixing here two styles, 
* passing op as an arg
* accessing c directly

i'd stick to one style, namely accessing directly, since it's the most common in our codebase.

so just 

```
for _, op := range ops {
  wg.Add(1)
  op := op
  go func() {
    // use op
  }()
```

i believe you can also copy `c := c` (along with `op:=op`, before the `go func`) and avoid the new name "cfgCopy" as well as `c = &cfgCopy`, but since it's a pointer i'm not 100% sure

Reply:
Yeah, agree. I think in that way is cleaner. :raised_hands:

Reply:
Updated